### ISSUE TITLE
NS_ERROR_ILLEGAL_VALUE exception in Firefox for large datasets

### ISSUE THREAD

**Issue by Ciki** [vojtech-dobes/history.nette.ajax.js#24]
[storing all snippets in cache](https://github.com/vojtech-dobes/history.nette.ajax.js/blob/master/client-side/history.ajax.js#L67) might cause the HTML5 pushstate cache to overrun Firefox's [640k character limit](https://developer.mozilla.org/en-US/docs/Web/Guide/API/DOM/Manipulating_the_browser_history#The_pushState()_method).

Might it be better to use localStorage or sessionStorage for this cache, as Mozilla's documentation suggests?

Similar issue [here](https://github.com/alfajango/jquery-dynatable/issues/80)

**Comment by lysergicfacet** [comment]
I agree with @Ciki .  Using localstorage by default and then falling back to the current method seems entirely reasonable.  If I wasn't bogged down with work, I would certainly help with this (as it's something that would make my job easier).  I am in full support of this getting solved.
